frontend/src/pages/storageClasses/StorageClassesTableRow.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+
+import {
+  Flex,
+  FlexItem,
+  Popover,
+  DescriptionList,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  DescriptionListDescription,
+  Tooltip,
+  Label,
+  Timestamp,
+} from '@patternfly/react-core';
+import { Tr, Td, ActionsColumn } from '@patternfly/react-table';
+import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+
+import { StorageClassConfig, StorageClassKind } from '~/k8sTypes';
+import { TableRowTitleDescription } from '~/components/table';
+import { FetchStateRefreshPromise } from '~/utilities/useFetchState';
+import { updateStorageClassConfig } from '~/services/StorageClassService';
+import DashboardPopupIconButton from '~/concepts/dashboard/DashboardPopupIconButton';
+import { NoValue } from '~/components/NoValue';
+import { ColumnLabel } from './constants';
+import { isOpenshiftDefaultStorageClass } from './utils';
+import { StorageClassEnableSwitch } from './StorageClassEnableSwitch';
+import { StorageClassDefaultRadio } from './StorageClassDefaultRadio';
+
+interface StorageClassesTableRowProps {
+  storageClass: StorageClassKind;
+  storageClassConfigMap: Record<string, StorageClassConfig | undefined>;
+  refresh: FetchStateRefreshPromise<StorageClassKind[]>;
+}
+
+export const StorageClassesTableRow: React.FC<StorageClassesTableRowProps> = ({
+  storageClass,
+  storageClassConfigMap,
+  refresh,
+}) => {
+  const storageClassConfig = storageClassConfigMap[storageClass.metadata.name];
```

Comment:
`storageClassConfigMap` is used elsewhere in this component, for instance to determine whether other rows are enabled as a part of logic to determine whether switches should be disabled. 

Since a row needs visibility into other row data, that's why this is here, and using the helper function `getStorageClassConfig` isn't required,  since that function is already called to compose `storageClassConfigMap`.

Reply:
yeah I ended up seeing that later in my review. this is fine then.